src/mixins/ldRedirect.js

```diff
@@ -53,14 +68,25 @@ export default (requiredFeatureFlag, to) => {
           this.ldRedirectFlagWatcher = null;
         }
       },
+      redirectHandler() {
+        if (this.$ld.ready && !this.ldRedirectFlagValue) {
+          this.$router.push(this.resolveRedirect);
+        } else if (!this.ldReady) {
+          this.setLdRedirectReadyWatcher();
+          this.setLdRedirectFlagWatcher();
+        }
+      },
     },
-    mounted() {
-      if (this.$ld.ready && !this.$ld.flags[requiredFeatureFlag || this.requiredFeatureFlag]) {
-        this.$router.push(to == null ? this.ldRedirectTo : to);
-      } else if (!this.ldReady) {
-        this.setLdRedirectReadyWatcher();
-        this.setLdRedirectFlagWatcher();
+    activated() {
+      if (this.hasBeenDeactivated) {
+        this.redirectHandler();
       }
     },
+    mounted() {
+      this.redirectHandler();
+    },
+    deactivated() {
+      this.hasBeenDeactivated = true;
+    },
```

Comment:
A note that this is intended to support components within <keep-alive> would be nice for future context.